src/_common/scroll/inview/inview.ts

```diff
@@ -1,76 +1,62 @@
 import Vue, { CreateElement } from 'vue';
-import { Component, Prop, Watch } from 'vue-property-decorator';
-import { arrayRemove } from '../../../utils/array';
-import { findVueParent } from '../../../utils/vue';
-import { Scroll } from '../scroll.service';
-import { ScrollInviewContainer } from './container';
+import { Component, Emit, Prop, Watch } from 'vue-property-decorator';
+import { findRequiredVueParent } from '../../../utils/vue';
 import { AppScrollInviewParent } from './parent';
 
-// This is the root container we use if there's no inview parent.
-let BaseContainer: ScrollInviewContainer;
-if (!GJ_IS_SSR) {
-	BaseContainer = new ScrollInviewContainer(Scroll.watcher);
-}
-
 @Component({})
 export class AppScrollInview extends Vue {
 	@Prop({ type: String, default: 'div' })
 	tag!: string;
 
-	@Prop({ type: Number, default: 0 })
-	extraPadding!: number;
-
+	/**
+	 * The margin will effectively get added to the bounding box of this element
+	 * to make it larger or smaller. It should be in the format of a CSS margin
+	 * property and always have the "px" after each value.
+	 */
+	@Prop({ type: String, default: '0px' })
+	margin!: string;
+
+	/**
+	 * The strict prop will cause the outview event to trigger when any part of
+	 * the element is out of view, instead of requiring the whole element to be
+	 * out of view.
+	 */
+	@Prop({ type: Boolean, default: false })
+	strict!: boolean;
+
+	// This will get set by AppScrollInviewContainer as this element goes into
+	// and out of view.
 	inView = false;
 
-	// Don't have Vue watch these by not setting their default values.
-	top!: number;
-	bottom!: number;
-
-	get container() {
-		const parent = findVueParent(this, AppScrollInviewParent);
-		return parent ? parent._container : BaseContainer;
+	private get parent() {
+		return findRequiredVueParent(this, AppScrollInviewParent);
 	}
 
-	async mounted() {
-		// Wait for the parent container to bootstrap in.
-		await this.$nextTick();
-		this.container.items.push(this);
-
-		// Queue a check so that we can group together all the other components that may mount in
-		// and do one single check.
-		this.container.queueCheck();
+	private get container() {
+		return this.parent.getContainer(this.margin);
```

Comment:
It's non-reactive. We have this pattern in the codebase quite a bit, where it initializes with the initial prop but then doesn't change if it ever updates. To support reactive will be pretty hard and not worth it, I think. Instead we should probably come up with some sort of convention to signify when something isn't reactive. We don't have that convention yet, so I didn't do anything for it.

Reply:
I think what we should do is read the prop value in mounted and set it to a variable that's named a bit more verbosely, like "initMargin". We should also add a comment on the prop because we can't convey this restriction at compile time.